### DISCUSSION TITLE
Emacs support for editing P files

### DISCUSSION THREAD

**Discussion by rudi** [p-org/P#693] [Feature Request!]
Hi, I implemented a quick-and-dirty emacs mode for P at https://github.com/rudi/p-mode.  It's about as featureful as the sublime mode that I used for inspiration.  Would be happy to have it hosted at p-org or to continue developing it on my own repo :)

**Comment by ankushdesai** [comment]
This is great! thanks for doing this. 
It would be great if we can make it feature complete and share it with others! 
I would be fine maintaining it at your repo and adding a link to P documentation.

**Comment by rudi** [comment]
Hi, that's great to hear!  The mode is usable as-is but I'll continue adding features as I use P.  Do let me know which features would be best to have!

I'm currently working on a menu for running the tools (pmc etc.) from within Emacs.  I'll likely wait with implementing automatic indentation until the tree-sitter support in Emacs has arrived (first half of next year), since a tree-sitter grammar will be usable in vim as well as far as I know.

**Comment by ankushdesai** [comment]
Our team is only planning to support Peasy as a framework.
